distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerProcessTests.java

```diff
@@ -277,82 +280,119 @@ public void testLibffiEnv() throws Exception {
+        assertThat(tmpDir.getParent().toString(), equalTo(baseTmpDir.toString()));
     }
 
     public void testTempDirOverride() throws Exception {
         Path customTmpDir = createTempDir();
         envVars.put("ES_TMPDIR", customTmpDir.toString());
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(tmpDir.toString(), equalTo(customTmpDir.toString()));
-            return new ArrayList<>();
-        };
+        var tmpDir = ServerProcessUtils.setupTempDir(createProcessInfo());
+        assertThat(tmpDir.toString(), equalTo(customTmpDir.toString()));
+
         processValidator = pb -> assertThat(pb.environment(), not(hasKey("ES_TMPDIR")));
         runForeground();
     }
 
-    public void testTempDirOverrideMissing() throws Exception {
+    public void testTempDirOverrideMissing() {
         Path baseDir = createTempDir();
         envVars.put("ES_TMPDIR", baseDir.resolve("dne").toString());
-        var e = expectThrows(UserException.class, () -> runForeground());
+        var e = expectThrows(UserException.class, () -> ServerProcessUtils.setupTempDir(createProcessInfo()));
         assertThat(e.exitCode, equalTo(ExitCodes.CONFIG));
         assertThat(e.getMessage(), containsString("dne] does not exist"));
     }
 
     public void testTempDirOverrideNotADirectory() throws Exception {
         Path tmpFile = createTempFile();
         envVars.put("ES_TMPDIR", tmpFile.toString());
-        var e = expectThrows(UserException.class, () -> runForeground());
+        var e = expectThrows(UserException.class, () -> ServerProcessUtils.setupTempDir(createProcessInfo()));
         assertThat(e.exitCode, equalTo(ExitCodes.CONFIG));
         assertThat(e.getMessage(), containsString("is not a directory"));
     }
 
     public void testCustomJvmOptions() throws Exception {
+        var serverArgs = createServerArgs(false, false);
+        var jvmOptionsFile = serverArgs.configDir().resolve("jvm.options");
+        var esConfigFile = serverArgs.configDir().resolve("elasticsearch.yml");
+        try {
+            Files.createDirectories(serverArgs.configDir());
+            Files.createFile(jvmOptionsFile);
+            Files.createFile(esConfigFile);
+        } catch (IOException ex) {
+            // File or directory already exists
+        }
+
         envVars.put("ES_JAVA_OPTS", "-Dmyoption=foo");
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(envOptions, equalTo("-Dmyoption=foo"));
-            return new ArrayList<>();
-        };
+
+        var jvmOptions = JvmOptionsParser.determineJvmOptions(serverArgs, createProcessInfo(), Path.of(".'"));
+        assertThat(jvmOptions, hasItem("-Dmyoption=foo"));
+
         processValidator = pb -> assertThat(pb.environment(), not(hasKey("ES_JAVA_OPTS")));
         runForeground();
     }
 
+    public void testCommandLineDistributionType() throws Exception {
```

Comment:
This seems like it should be in JvmOptionsParserTests since it is checking the "parser" includes the distribution type, which the ServerProcess here knows nothing about.

Reply:
Moved!